privacy-on-beam/pbeam/aggregations.go

```diff
@@ -450,6 +483,213 @@ func clampNegativePartitionsFloat64Fn(v beam.V, r float64) (beam.V, float64) {
 func convertFloat32ToFloat64Fn(z beam.Z, f float32) (beam.Z, float64) {
 	return z, float64(f)
 }
+
 func convertFloat64ToFloat64Fn(z beam.Z, f float64) (beam.Z, float64) {
 	return z, f
 }
+
+// Turn PCollection<V> into PCollection <V,0>.
+func newPrepareAddPartitionsFn(vKind reflect.Kind) interface{} {
+	var err error
+	var fn interface{}
+	switch vKind {
+	case reflect.Int64:
+		fn = prepareAddPartitionsInt64Fn
+	case reflect.Float64:
+		fn = prepareAddPartitionsFloat64Fn
+	default:
+		log.Exitf("pbeam.newPrepareAddPartitionsFn: vKind(%v) should be int64 or float64", vKind)
+	}
+
+	if err != nil {
+		log.Exit(err)
+	}
+	return fn
+}
+
+// Turn PCollection<V> into either PCollection <V,[]int64{0}> or PCollection <V,[]float64{0}>.
+func newPrepareAddMeanPartitionsFn(vKind reflect.Kind) interface{} {
+	var err error
+	var fn interface{}
+	switch vKind {
+	case reflect.Int64:
+		fn = prepareAddPartitionsMeanInt64Fn
+	case reflect.Float64:
+		fn = prepareAddPartitionsMeanFloat64Fn
+	default:
+		log.Exitf("pbeam.newPrepareAddPartitionsFn: vKind(%v) should be int64 or float64", vKind)
+	}
+
+	if err != nil {
+		log.Exit(err)
+	}
+	return fn
+}
+
+func prepareAddPartitionsInt64Fn(partitionKey beam.X) (k beam.X, v int64) {
+	return partitionKey, 0
+}
+
+func prepareAddPartitionsFloat64Fn(partitionKey beam.X) (k beam.X, v float64) {
+	return partitionKey, 0
+}
+
+func prepareAddPartitionsMeanInt64Fn(partitionKey beam.X) (k beam.X, v []int64) {
+	return partitionKey, []int64{}
+}
+
+func prepareAddPartitionsMeanFloat64Fn(partitionKey beam.X) (k beam.X, v []float64) {
+	return partitionKey, []float64{}
+}
+
+
+type Partition struct {
+	PartitionKey string
+}
+
+// Function for sum and mean. Drop unspecified partitions.
+func correctPartitions(s beam.Scope, partitions []beam.PCollection, pcol PrivatePCollection, partitionT beam.EncodedType) beam.PCollection {
+	if len(partitions) == 1 {
+		partitionsCol := partitions[0]
+		partitionsMap := beam.Combine(s, newKVPartitionsHashMapFn(partitionT), partitionsCol)
+		return beam.ParDo(s, prunePartitionsFn, pcol.col, beam.SideInput{Input: partitionsMap})
+	}
+	return pcol.col
+}
+
+// Function for count and distinct_id. Drop unspecified partitions.
+func correctCountPartitions(s beam.Scope, partitions []beam.PCollection, pcol PrivatePCollection, partitionT typex.FullType) beam.PCollection {
```

Comment:
Similarly here, can we call this something like DropUnspecifiedPartitionsForCount?